src/reanimated2/animation/spring.ts

```diff
@@ -38,44 +55,90 @@ export function withSpring(
 
   return defineAnimation<SpringAnimation>(toValue, () => {
     'worklet';
-
-    // TODO: figure out why we can't use spread or Object.assign here
-    // when user config is "frozen object" we can't enumerate it (perhaps
-    // something is wrong with the object prototype).
-    const config: Required<SpringConfig> = {
+    const defaultConfig: Record<keyof SpringConfig, any> = {
       damping: 10,
       mass: 1,
       stiffness: 100,
       overshootClamping: false,
       restDisplacementThreshold: 0.01,
       restSpeedThreshold: 2,
       velocity: 0,
-    };
-    if (userConfig) {
-      Object.keys(userConfig).forEach(
-        (key) =>
-          ((config as any)[key] = userConfig[key as keyof typeof userConfig])
-      );
+      duration: 2000,
+      dampingRatio: 0.5,
+    } as const;
+
+    const config = { ...defaultConfig, ...userConfig };
+
+    function initialCalculations(mass = 0): {
```

Comment:
The below methods all seem to not rely on the fact they are defined in the animation method scope. If that's the case we should extract them to the file scope and add `'worklet'`, or put as many things as we can in springUtils.ts

Reply:
All of them use in-scope variables, so I am not sure if it will really look better after extracting it.